### PR TITLE
#10560 - Differentiate between sensitive and personal pseudonymizatio…

### LINKED ISSUE #10560: [Pseudonymization] Mark UI inputs as confidential depending on what rights the user has
### Problem Description

There are separate user rights for seeing personal and sensitive data, on the backend DTO-s are pseudonymized based on those right, so no sensitive data will be pseudonymized if the user has right for seeing sensitive data but no right for personal data.
On the UI (web and mobile) all personal and  sensitive fields are marked as confidential even when the user should se sensitive data but not the personal ones
### Proposed Change

User should see and should be able to edit sensitive data if it has right to see sensitive data but no right to see personal data.
### Implementation Details

In the UI code combine the `Pseudonymizable::isPseudonymized` method and checking whit right the current user has, when creating `UiFieldAccessCheckers` instances 
### Additional Information

Please do it for web and mobile client too

### QA Information
Please check all forms in the web app with a user that has only sensitive/personal access to data; if they have only access to sensitive data, all fields that are only annotated with `@PersonalData` (e.g. person first and last name) should be pseudonymized; if they have only access to personal data, all fields that are only annotated with `@SensitiveData` (e.g. the additional comments of cases) should be pseudonymized. Up until now, there was no proper differentiation between the two, so a user with no access to personal, but access to sensitive data would still see the sensitive data fields being pseudonymized and read only in the UI. Also, there's a differentiation between data IN the user's jurisdiction and OUTSIDE the user's jurisdiction depending on the user role configuration.

### PR DESCRIPTION
**Pull request by MateStrysewske**
…n in UI

Fixes #10560

### CODE REVIEW

**Review by BarnaBartha — COMMENTED** [review]
i would like to see for at least one dto for example CaseDataDto or PersonDto that these changes are covered with tests
that is asserting what a user would see now considering the rights:
	SEE_PERSONAL_DATA_IN_JURISDICTION(UserRightGroup.DATA_PROTECTION),
	SEE_PERSONAL_DATA_OUTSIDE_JURISDICTION(UserRightGroup.DATA_PROTECTION),
	SEE_SENSITIVE_DATA_IN_JURISDICTION(UserRightGroup.DATA_PROTECTION),
	SEE_SENSITIVE_DATA_OUTSIDE_JURISDICTION(UserRightGroup.DATA_PROTECTION),

File: sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/ClinicalCourseController.java [hunk 1]
```diff
@@ -35,7 +35,8 @@ public void openClinicalVisitCreateForm(ClinicalCourseReferenceDto clinicalCours
 			true,
 			clinicalVisit.getDisease(),
 			FacadeProvider.getPersonFacade().getByUuid(caze.getPerson().getUuid()),
-			false);
+			false,
+			true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/ClinicalCourseController.java:39]: Maybe add a comment why setting this values directly is valid

File: sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactController.java [hunk 2]
```diff
@@ -818,7 +818,12 @@ public void openSelectCaseForContactWindow(Disease disease, Consumer<CaseSelecti
 	public CommitDiscardWrapperComponent<EpiDataForm> getEpiDataComponent(final String contactUuid) {
 
 		ContactDto contact = FacadeProvider.getContactFacade().getByUuid(contactUuid);
-		EpiDataForm epiDataForm = new EpiDataForm(contact.getDisease(), ContactDto.class, contact.getEpiData().isPseudonymized(), null);
+		EpiDataForm epiDataForm = new EpiDataForm(
+			contact.getDisease(),
+			ContactDto.class,
+			contact.getEpiData().isPseudonymized(),
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactController.java:824]: Extract var

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java [hunk 3]
```diff
@@ -591,7 +591,7 @@ public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponent(Case
 			caseDataDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(caseRef.getUuid());
 		}
 
-		EventDataForm eventCreateForm = new EventDataForm(true, false);
+		EventDataForm eventCreateForm = new EventDataForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java:594]: Again it is very hard for an outsider to understand the reasoning why we set these values always to a certain value. I would add a short comment.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java [hunk 4]
```diff
@@ -631,7 +631,7 @@ public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponentForCa
 		List<CaseDataDto> caseDataDtos =
 			FacadeProvider.getCaseFacade().getByUuids(caseRefs.stream().map(c -> c.getUuid()).collect(Collectors.toList()));
 
-		EventDataForm eventCreateForm = new EventDataForm(true, false);
+		EventDataForm eventCreateForm = new EventDataForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java:634]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java [hunk 5]
```diff
@@ -658,7 +658,7 @@ public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponentForCo
 		List<ContactDto> contactDtos =
 			FacadeProvider.getContactFacade().getByUuids(contactRefs.stream().map(c -> c.getUuid()).collect(Collectors.toList()));
 
-		EventDataForm eventCreateForm = new EventDataForm(true, false);
+		EventDataForm eventCreateForm = new EventDataForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java:661]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java [hunk 6]
```diff
@@ -682,7 +682,7 @@ public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponentForCo
 
 	public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponent(ContactDto contact) {
 
-		EventDataForm eventCreateForm = new EventDataForm(true, false);
+		EventDataForm eventCreateForm = new EventDataForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java:685]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java [hunk 7]
```diff
@@ -712,7 +712,7 @@ public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponent(
 		boolean createSuperordinateEvent) {
 
 		EventDto superOrSubordinateEvent = FacadeProvider.getEventFacade().getEventByUuid(superOrSubordinateEventRef.getUuid(), false);
-		EventDataForm form = new EventDataForm(true, false);
+		EventDataForm form = new EventDataForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java:715]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/labmessage/LabMessageProcessingFlow.java [hunk 8]
```diff
@@ -261,7 +261,13 @@ protected void handleCreateEventParticipant(
 		HandlerCallback<EventParticipantDto> callback) {
 		Window window = VaadinUiUtil.createPopupWindow();
 
-		EventParticipantEditForm createForm = new EventParticipantEditForm(event, false, eventParticipant.getPerson().isPseudonymized(), false);
+		EventParticipantEditForm createForm = new EventParticipantEditForm(
+			event,
+			false,
+			true,
+			eventParticipant.getPerson().isPseudonymized(),
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/labmessage/LabMessageProcessingFlow.java:268]: This ordering looks odd, until now, it jurisdiction flag came always after pseudonomization, now there is a bool inbetween

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java [hunk 9]
```diff
@@ -84,7 +84,7 @@ public CommitDiscardWrapperComponent<PathogenTestForm> getPathogenTestCreateComp
 		int caseSampleCount,
 		BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest,
 		boolean suppressNavigateToCase) {
-		PathogenTestForm createForm = new PathogenTestForm(sampleDto, true, caseSampleCount, false);
+		PathogenTestForm createForm = new PathogenTestForm(sampleDto, true, caseSampleCount, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java:87]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java [hunk 10]
```diff
@@ -101,13 +102,15 @@ public class PathogenTestForm extends AbstractEditForm<PathogenTestDto> {
...
 
-	public PathogenTestForm(SampleDto sample, boolean create, int caseSampleCount, boolean isPseudonymized) {
+	public PathogenTestForm(SampleDto sample, boolean create, int caseSampleCount, boolean isPseudonymized, boolean inJurisdiction) {
 		super(
 			PathogenTestDto.class,
 			PathogenTestDto.I18N_PREFIX,
 			false,
 			FieldVisibilityCheckers.withDisease(null).andWithCountry(FacadeProvider.getConfigFacade().getCountryLocale()),
-			UiFieldAccessCheckers.forSensitiveData(!create && isPseudonymized));
+			UiFieldAccessCheckers.forDataAccessLevel(
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java:111]: add comment explaining the or and and in the method calls

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleController.java [hunk 11]
```diff
@@ -173,7 +173,8 @@ public PathogenTestForm addPathogenTestComponent(
 		horizontalRule.setWidth(100f, Unit.PERCENTAGE);
 		sampleComponent.addComponent(horizontalRule, sampleComponent.getComponentCount() - 1);
 
-		PathogenTestForm pathogenTestForm = new PathogenTestForm(sampleComponent.getWrappedComponent().getValue(), true, caseSampleCount, false);
+		PathogenTestForm pathogenTestForm =
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleController.java:176]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/therapy/TherapyController.java [hunk 12]
```diff
@@ -40,7 +40,7 @@ public TherapyController() {
 	}
 
 	public void openPrescriptionCreateForm(TherapyReferenceDto therapy, Runnable callback) {
-		PrescriptionForm form = new PrescriptionForm(true, false, false);
+		PrescriptionForm form = new PrescriptionForm(true, false, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/therapy/TherapyController.java:43]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/therapy/TherapyController.java [hunk 13]
```diff
@@ -170,7 +168,7 @@ public void onCommit() {
 	}
 
 	public void openTreatmentCreateForm(PrescriptionDto prescription, Runnable callback) {
-		TreatmentForm form = new TreatmentForm(true, false);
+		TreatmentForm form = new TreatmentForm(true, false, true);
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/therapy/TherapyController.java:171]: same here

File: sormas-ui/src/main/java/de/symeda/sormas/ui/vaccination/list/VaccinationList.java [hunk 14]
```diff
@@ -75,7 +76,9 @@ protected void drawDisplayedEntries() {
 						.edit(
 							FacadeProvider.getVaccinationFacade().getByUuid(listEntry.getVaccination().getUuid()),
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/vaccination/list/VaccinationList.java:77]: replace `getVaccination` with var

File: sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java [hunk 15]
```diff
@@ -88,12 +103,12 @@ public VisitEditForm(Disease disease, ContactDto contact, CaseDataDto caze, Pers
 
 	}
 
-	public VisitEditForm(Disease disease, ContactDto contact, PersonDto person, boolean create, boolean isInJurisdiction) {
-		this(disease, contact, null, person, create, contact.isPseudonymized());
+	public VisitEditForm(Disease disease, ContactDto contact, PersonDto person, boolean create) {
+		this(disease, contact, null, person, create, contact.isPseudonymized(), contact.isInJurisdiction());
```
Comment (JonasCir) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java:107]: In the old version of `VisitController.java` there was a negation. Double check this.

**Review by JonasCir — CHANGES_REQUESTED** [review]
Comments are mostly about extracting vars to avoid multiple getter calls. 

Also I noticed that even though I'm familiar with the project, I have a hard time to reason why it is justified in some cases to set the pseudonymization and jurisdiction bool directly. I pointed a couple of calls out where I think it helps the reader to understand why setting the values is justified.

**Comment by MateStrysewske** [comment]
> i would like to see for at least one dto for example CaseDataDto or PersonDto that these changes are covered with tests that is asserting what a user would see now considering the rights: SEE_PERSONAL_DATA_IN_JURISDICTION(UserRightGroup.DATA_PROTECTION), SEE_PERSONAL_DATA_OUTSIDE_JURISDICTION(UserRightGroup.DATA_PROTECTION), SEE_SENSITIVE_DATA_IN_JURISDICTION(UserRightGroup.DATA_PROTECTION), SEE_SENSITIVE_DATA_OUTSIDE_JURISDICTION(UserRightGroup.DATA_PROTECTION),

@BarnaBartha The problem is that I believe the data was already correctly passed to the UI before, what I changed is that it's now also properly shown in the UI, which I can't test with unit tests; however, I can still add a test that confirms that there's a correct differentiation between sensitive and personal pseudonymization in the backend (and has been before).

File: sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/labmessage/LabMessageProcessingFlow.java (hunk 8 above)
Comment (MateStrysewske) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/labmessage/LabMessageProcessingFlow.java:268]: It still comes after the pseudonymized flag, but there are two different pseudonymization flags in this method (one for the event participant, one for its person) and I actually mixed something up here. Good catch, fixed it.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java (hunk 15 above)
Comment (MateStrysewske) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java:107]: Yes, that's fine, the previous logic basically assumed that whenever an entry is not pseudonymized, it's automatically in the user's jurisdiction, which is technically wrong.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java [hunk 16]
```diff
@@ -88,12 +103,12 @@ public VisitEditForm(Disease disease, ContactDto contact, CaseDataDto caze, Pers
...
 	}
 
-	public VisitEditForm(Disease disease, CaseDataDto caze, PersonDto person, boolean create, boolean isInJurisdiction) {
-		this(disease, null, caze, person, create, caze.isPseudonymized());
+	public VisitEditForm(Disease disease, CaseDataDto caze, PersonDto person, boolean create) {
+		this(disease, null, caze, person, create, caze.isPseudonymized(), caze.isInJurisdiction());
```
[the remark this answers is not in the document]
Comment (MateStrysewske) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java:111]: see above

**Review by BarnaBartha — APPROVED** [review]

**Review by JonasCir — APPROVED** [review]
Thanks @MateStrysewske !

**Pull request merged**